db/schema.rb

```diff
@@ -311,17 +311,19 @@
   end
 
   create_table "course_material_folders", force: :cascade do |t|
-    t.integer  "parent_folder_id", index: {name: "fk__course_material_folders_parent_folder_id"}, foreign_key: {references: "course_material_folders", name: "fk_course_material_folders_parent_folder_id", on_update: :no_action, on_delete: :no_action}
     t.integer  "owner_id"
-    t.string   "owner_type",       limit: 255, index: {name: "fk__course_material_folders_owner_id", with: ["owner_id"]}
-    t.string   "name",             limit: 255, null: false
+    t.string   "owner_type",         limit: 255, index: {name: "fk__course_material_folders_owner_id", with: ["owner_id"]}
+    t.integer  "course_id",          null: false, index: {name: "fk__course_material_folders_course_id"}, foreign_key: {references: "courses", name: "fk_course_material_folders_course_id", on_update: :no_action, on_delete: :no_action}
+    t.integer  "parent_id",          index: {name: "fk__course_material_folders_parent_id"}, foreign_key: {references: "course_material_folders", name: "fk_course_material_folders_parent_id", on_update: :no_action, on_delete: :no_action}
+    t.string   "name",               limit: 255,                 null: false
```

Comment:
this should be unique for every parent? the unique key should be (parent_id, name)